core/src/test/java/org/apache/iceberg/TestPartitionSpecInfo.java

```diff
@@ -91,25 +94,27 @@ public void testSpecInfoPartitionedTable() {
   }
 
   @Test
-  public void testSpecInfoPartitionSpecEvolutionForV1Table() {
+  public void testSpecInfoPartitionSpecEvolution() {
     PartitionSpec spec = PartitionSpec.builderFor(schema)
         .bucket("data", 4)
         .build();
     TestTables.TestTable table = TestTables.create(tableDir, "test", schema, spec, formatVersion);
 
     Assert.assertEquals(spec, table.spec());
 
-    TableMetadata base = TestTables.readMetadata("test");
-    PartitionSpec newSpec = PartitionSpec.builderFor(table.schema())
-        .bucket("data", 10)
+    PartitionSpec expectedNewSpec = PartitionSpec.builderFor(table.schema())
+        .add(2, newFieldId, "data_bucket", "bucket[10]")
         .withSpecId(1)
         .build();
-    table.ops().commit(base, base.updatePartitionSpec(newSpec));
 
-    Assert.assertEquals(newSpec, table.spec());
-    Assert.assertEquals(newSpec, table.specs().get(newSpec.specId()));
+    table.updateSpec().clear()
```

Comment:
Using `clear` should be rare. Most cases should remove a single field and add another field.